pkg/server/router/credential.go

```diff
@@ -145,6 +147,140 @@ func (cr CredentialRouter) GetCredential(ctx context.Context, w http.ResponseWri
 	return framework.Respond(ctx, w, resp, http.StatusOK)
 }
 
+type GetCredentialStatusResponse struct {
+	Revoked bool `json:"revoked"`
+}
+
+// GetCredentialStatus godoc
+// @Summary      Get Credential Status
+// @Description  Get credential status by id
+// @Tags         CredentialAPI
+// @Accept       json
+// @Produce      json
+// @Param        id   path      string  true  "ID"
+// @Success      200  {object}  GetCredentialStatusResponse
+// @Failure      400  {string}  string  "Bad request"
+// @Router       /v1/credentials/{id}/status [get]
+func (cr CredentialRouter) GetCredentialStatus(ctx context.Context, w http.ResponseWriter, _ *http.Request) error {
+	id := framework.GetParam(ctx, IDParam)
+	if id == nil {
+		errMsg := "cannot get credential without ID parameter"
+		logrus.Error(errMsg)
+		return framework.NewRequestErrorMsg(errMsg, http.StatusBadRequest)
+	}
+
+	gotCredential, err := cr.service.GetCredential(credential.GetCredentialRequest{ID: *id})
+	if err != nil {
+		errMsg := fmt.Sprintf("could not get credential with id: %s", *id)
+		logrus.WithError(err).Error(errMsg)
+		return framework.NewRequestError(errors.Wrap(err, errMsg), http.StatusBadRequest)
+	}
+
+	resp := GetCredentialStatusResponse{
+		Revoked: gotCredential.Revoked,
+	}
+
+	return framework.Respond(ctx, w, resp, http.StatusOK)
+}
+
+type GetCredentialStatusListResponse struct {
+	ID            string                        `json:"id"`
```

Comment:
The credential object already has a field called `ID`. What's the reasoning behind including both?

Reply:
if the credential comes back as a JWT this lets the caller know the ID without needing to decode the JWT